packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -815,17 +816,41 @@ export function usePress(props: PressHookProps): PressResult {
 
   // Avoid onClick delay for double tap to zoom by default.
   useEffect(() => {
-    let element = domRef?.current;
-    if (element && (element instanceof getOwnerWindow(element).Element)) {
-      // Only apply touch-action if not already set by another CSS rule.
-      let style = getOwnerWindow(element).getComputedStyle(element);
-      if (style.touchAction === 'auto') {
-        // touchAction: 'manipulation' is supposed to be equivalent, but in 
-        // Safari it causes onPointerCancel not to fire on scroll.
-        // https://bugs.webkit.org/show_bug.cgi?id=240917
-        (element as HTMLElement).style.touchAction = 'pan-x pan-y pinch-zoom';
+    if (!domRef || process.env.NODE_ENV === 'test') {
+      return;
+    }
+
+    const ownerDocument = getOwnerDocument(domRef.current);
+    if (!ownerDocument || !ownerDocument.head) {
+      return;
+    }
+
+    if (ownerDocument.getElementById(styleId)) {
+      return;
+    }
+
+    // Skip adding if meta viewport tag has width=device-width
+    const viewportMeta = ownerDocument.querySelector('meta[name="viewport"]');
+    if (viewportMeta) {
+      const content = viewportMeta.getAttribute('content');
+      if (content && content.includes('width=device-width')) {
+        return;
       }
     }
+
+    const style = ownerDocument.createElement('style');
+    style.id = styleId;
+    // touchAction: 'manipulation' is supposed to be equivalent, but in
+    // Safari it causes onPointerCancel not to fire on scroll.
+    // https://bugs.webkit.org/show_bug.cgi?id=240917
+    style.textContent = `
+@layer {
+  [data-react-aria-pressable] {
```

Comment:
```suggestion
  [${STYLE_ID}] {
```

Reply:
and i just realized these are different, lets move them out to a constant as well

Reply:
These area different strings, but I can do two consts.